tests/functional/adapter/catalog_tests/test_relation_types.py

```diff
@@ -0,0 +1,44 @@
+from dbt.contracts.results import CatalogArtifact
+from dbt.tests.util import run_dbt
+import pytest
+
+from tests.functional.adapter.catalog_tests import files
+
+
+class TestCatalogRelationTypes:
+    @pytest.fixture(scope="class", autouse=True)
+    def seeds(self):
+        return {"my_seed.csv": files.MY_SEED}
+
+    @pytest.fixture(scope="class", autouse=True)
+    def models(self):
+        yield {
```

Comment:
What's up with `yield` over return.

Also, really like the flow this test. Points for aesthetic.

Reply:
I suppose it doesn't matter here, as nothing happens after the `yield` in the fixture. The convention for a pytest fixture is:

```python
@pytest.fixture()
def my_cool_fixture(self):
    # do some setup pre-test
    yield some_value_you_want
    # do some teardown post-test
```

When the fixture simply returns a value, this can be simplified to:

```python
@pytest.fixture()
def my_cool_fixture(self):
    return some_value_you_want
```

I use `yield` out of habit of writing a pytest fixture, but both are valid.

Reply:
Wouldn't have guessed they were functionally equivalent. Weird then that we are using return in several places where yield does the job (and is more idiomatic). Oh well -- getting to know our tools! :)